src/ossl/common.rs

```diff
@@ -261,6 +286,23 @@ impl OsslParam {
         Ok(self)
     }
 
+    pub fn add_int(
+        mut self,
+        key: *const c_char,
+        val: usize,
+    ) -> KResult<OsslParam> {
+        if self.finalized {
+            return err_rv!(CKR_GENERAL_ERROR);
+        }
+
+        let container = (val as c_int).to_ne_bytes().to_vec();
+        unsafe {
+            OSSL_PARAM_construct_int(key, container.as_ptr() as *mut c_int);
+        }
+        self.v.push(container);
```

Comment:
Here is the same bug I am fixing in https://github.com/latchset/kryoptic/pull/39/commits/3baad23f27414d0a43f2574f289bd768ad9746b0
```suggestion
        let param = unsafe {
            OSSL_PARAM_construct_int(key, container.as_ptr() as *mut c_int)
        };
        self.v.push(container);
        self.v.push(param);
```
This compiles ok, but just throws away the parameter we created.

The above mentioned PR has fix for `size_t` case already.

Reply:
Ok I will merge your PR first and then rebase this code on top of it and fix the conflict.